File: app/src/main/java/com/mapelli/simone/githubclient/data/DateConverter.java
package com.mapelli.simone.githubclient.data;

import android.util.Log;

import java.util.Date;

import androidx.room.TypeConverter;

public class DateConverter {
    private final static String TAG = DateConverter.class.getSimpleName();

    @TypeConverter
    public static Date toDate(Long timestamp) {
        Log.d(TAG, "to Date from milllisec : " + timestamp + " is " + new Date(timestamp));
        return timestamp == null ? null : new Date(timestamp);

    }

    @TypeConverter
    public static Long fromDate(Date date) {
        Log.d(TAG, "to millisecfromepoch from date: " + date + " is " + date.getTime());
        return date == null ? null : date.getTime();

    }
}

File: app/src/main/java/com/mapelli/simone/githubclient/data/DateConverter.kt
package com.mapelli.simone.githubclient.data

import android.util.Log

import androidx.room.TypeConverter
import java.util.*

object DateConverter {
    private val TAG = DateConverter::class.java.getSimpleName()

    @TypeConverter
    fun longToCalendar(timestamp: Long): Calendar? {
        Log.d(TAG, "to Date from milllisec : " + timestamp + " is " + Date(timestamp!!))
        // return if (timestamp == null) null else Date(timestamp)
        return Calendar.getInstance().apply { timeInMillis = timestamp }


    }

    @TypeConverter
    fun calendarToLong(calendar: Calendar): Long? {
        Log.d(TAG, "to millisecfromepoch from date: " + calendar + " is "
                + calendar?.timeInMillis)
        // return date?.time
        return calendar?.timeInMillis
    }
}
